prow/kube/client.go

```diff
@@ -301,6 +306,45 @@ func NewClientFromFile(clusterPath, namespace string) (*Client, error) {
 	return NewClient(&c, namespace)
 }
 
+// TODO: add test that validates that different config formats cannot be confused.
+
+// ClientMapFromFile reads the file at clustersPath and attempts to load a map of cluster aliases
+// to authenticated clients to the respective clusters.
+// The file at clustersPath is expected to be a yaml map from strings to Cluster structs OR it may
+// simply be a single Cluster struct which will be assigned the alias $DefaultClusterAlias.
+// If the file is an alias map, it must include the alias $DefaultClusterAlias.
+func ClientMapFromFile(clustersPath, namespace string) (map[string]*Client, error) {
+	data, err := ioutil.ReadFile(clustersPath)
+	if err != nil {
+		return nil, err
+	}
+	var raw map[string]Cluster
+	if err := yaml.Unmarshal(data, &raw); err != nil {
+		// If we failed to unmarshal the multicluster format try the single Cluster format.
```

Comment:
That feels a *little* bit icky. Is anyone else using a separate build cluster? We might be able to get away with breaking the format here.

Reply:
I'm not sure if anyone else is using a separate build cluster. It doesn't look like openshift or istio do, but I figured it was better to be backwards compatible with existing deployments since it only adds a few lines of code.
Is there an advantage to requiring the new format?

Reply:
cc: @munnerz

Reply:
> Is there an advantage to requiring the new format?

It's just simpler to document "put it in this format" instead of "put it in this format, but also this other format will work," and the code is also simpler.

Reply:
One other thing to note is that we will still need to support the old format in the "cluster" config files (as opposed to the "build cluster" config files) because those files only specify a single cluster. 
Allowing the same format to be used in both files might be more important than minimizing complexity? I don't feel strongly about this though, I can remove support for the old format if that is still preferred.

Reply:
AFAICT we can always document the new format and remove backwards compatibility in a follow-up if we so desire.